### ISSUE TITLE
Something strange with hashcode

### ISSUE THREAD

**Issue by Renesanse** [felangel/equatable#52]
code: 
```dart
void main() {
  test('fits', () {
    final test = Test('');
    final test2 = Test('123');
    print(test.hashCode);
    print(test2.hashCode);
  });
}

class Test extends Equatable {
  final id;
  final items = [];

  Test(this.id);

  @override
  List<Object> get props => [id, items];
}
```

Expected: different hashcodes
Actual: the same

code: 
```dart
void main() {
  test('fits', () {
    final test = Test('');
    final test2 = Test('123');
    print(test.hashCode);
    print(test2.hashCode);
  });
}

class Test extends Equatable {
  var id = '';
  final items = [];

  @override
  //order of props changed
  List<Object> get props => [items, id];
}
```
Expected: different hashcodes
Actual: different hashcodes

**Comment by Renesanse** [comment]
I solved my problem by ... operator from this issue: https://github.com/felangel/equatable/issues/49. But,  my example really looks strange, what possibly could be wrong?

**Comment by felangel** [comment]
Hi @Renesanse 👋 
Thanks for opening an issue! I believe this issue has been fixed in #49 and published in v1.0.2

**Comment by Renesanse** [comment]
Thx =)
